src/parameterizations/lateral/MOM_thickness_diffuse.F90

```diff
@@ -1566,8 +1613,23 @@ subroutine thickness_diffuse_full(h, e, Kh_u, Kh_v, tv, uhD, vhD, cg1, dt, G, GV
     if (.not. CS%GM_src_alt) then ; if (allocated(MEKE%GM_src)) then
       MEKE%GM_src(i,j) = MEKE%GM_src(i,j) + Work_h
     endif ; endif
+    if (skeb_use_gm) then
+      h_tot = sum(h(i,j,1:nz))
+      skeb_gm_work(i,j)   = STOCH%skeb_gm_coef * Work_h
+      skeb_ebt_norm2(i,j) = GV%H_to_RZ * &
+                                     (sum(h(i,j,1:nz) * VarMix%ebt_struct(i,j,1:nz)**2) + h_neglect)
```

Comment:
In the past, we have found the use of the Fortran `sum()` function to be problematic when it is applied to real arrays because it leaves the order of arithmetic up to the compiler (and hence can never be safely refactored) and because it adds an implicit non-stride-1 do loop in the vertical.  Please see https://github.com/mom-ocean/MOM6/wiki/Code-style-guide#arithmetic-reproducibility for a more complete discussion.

Please consider revising these new additions to avoid the use of the `sum()` function.